bin/http-signature-header-js

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env node
+
+const program = require('commander');
+const httpSigs = require('../lib/');
+const fs = require('fs');
+const util = require('util');
+const httpMessageParser = require('http-message-parser');
+
+// this can be use to await paths to private keys and other files.
+// also can get be used to dereference keyId in middleware
+const readFile = util.promisify(fs.readFile);
+
+/**
+ * Simple wrapper around node's process.stdin.
+ * This allows us to get the result of cat and other commands.
+ *
+ * @param {string} [encoding='utf8'] - The http message's encoding.
+ *
+ * @returns {Promise<string>} Should return the http message.
+ */
+async function getStdIn(encoding = 'utf8') {
+  const {stdin} = process;
+  let message = '';
+  return new Promise((resolve, reject) => {
+    try {
+      stdin.setEncoding(encoding);
+      stdin.on('readable', () => {
+        let chunk;
+        while((chunk = stdin.read())) {
+          message += chunk;
+        }
+      });
+      stdin.on('end', () => resolve(message));
+    }
+    catch(e) {
+      reject(e);
+    }
+  });
+}
+
+async function getHttpMessage() {
+  const httpMessage = await getStdIn();
+  if(!httpMessage) {
+    throw new Error('An Http Message must be passed to stdin.');
+  }
+  // this will create a request or response object
+  // similar to node's default request object.
+  const request = httpMessageParser(httpMessage);
+  return request;
+}
+
+function makeHttpMessageOutgoing({status = '200 OK', version = '1.1', headers = {}}) {
```

Comment:
This should probably only be formatHttpHeaders or produceHttpMessage... you don't need '200 OK', you're not simulating an HTTP Response. You are taking input and then modifying to send to a server.